src/edu/buffalo/cse/jive/finiteStateMachine/models/TransitionBuilder.java

```diff
@@ -56,21 +58,54 @@ private void addColorTransition(State state1, State state2, String color) {
 		this.transitions.append(s);
 		addNewLine();
 	}
+	
+	private void addNoColorTransition(State state1, State state2) {
+		String s = "\"" + state1.toString() + "\"" + " --> " + "\"" + state2.toString() + "\"";
+		this.transitions.append(s);
+		addNewLine();
+	}
+	
+	private void addColorTransitionWithArrowBetweenSameStates(State state1, State state2, String backgroundColor, String arrowColor) {
+		
+		Pair<State, State> pair = new Pair<State, State>(state1, state2);
+		
+		String s = "\"" + state1.toString() + "\"";
+		if(ShortestPathHolderForEExpression.getPath().contains(pair)) 
+			s+= " -[#" + arrowColor + "]-> ";
+		else s += " --> ";
+		s += "\"" + state2.toString() + "\"" + " #" + backgroundColor;
+		this.transitions.append(s);
+		addNewLine();
+	}
+

```

Comment:
The method addColorTransitionWithArrowBetweenSameStates colors the target state as green. If the **pair of states belong to the shortest path** is part of the shortest path we found, it puts green arrows between them.